twitter_clone/main.py

```diff
@@ -21,9 +30,159 @@ def login_required(f):
+            cursor = g.db.execute(query, (username, password))
+            # cursor = g.db.execute(query, (username, hashedPassword))
+            results = cursor.fetchall()
+        # if results == None: # not this
+        #return username + password + str(results)  #empty list???
+            user_id = results[0][0]
+            username = results[0][1]
+            session["logged_in"] = True
+            session["user_id"] = str(user_id)
+            session["username"] = username
+            return redirect(url_for('display_feed', username = session['username']))
+        except: # here
+            
+            return redirect(url_for('login'))
+
+def _is_user_page(username):
+    """
+    Checks to see if the user is visiting his/her own timeline
+    """
+    return session['username'] == username
+
+@app.route("/<username>", methods = ["GET", "POST"])
+@login_required
+def display_feed(username):
+    
+    if _is_user_page(username) == True:
+        
+        if request.method == 'GET':
+            tweets = _retrieve_tweets(session['user_id'])
+            return render_template('own_feed.html', tweets=tweets)
+            
+        if request.method == 'POST':
+            #check if tweet is less than 140 chars, otherwise spit a message
+            _post_tweet(session['user_id'], request.form['tweet'])
+            # want to upload request.form['tweet'] = tweet text, need user_id that posted it
+            tweets = _retrieve_tweets(session['user_id'])
+            return render_template('own_feed.html', tweets=tweets)
+            
+    elif _is_user_page(username) == False:
+        if request.method == 'GET':
+            user_id = _get_user_id(username)
+            tweets = _retrieve_tweets(user_id)
+            return render_template('other_feed.html', tweets=tweets, username=username)
+
+@app.route("/tweets/<tweet_id>/delete", methods = ["POST"])    
+def delete(tweet_id):
+    _delete_tweet(tweet_id)
+    return redirect(url_for('own_feed', username = session['username']))
+
+@app.route("/logout/")
+def logout():
+    session.clear()
+    return redirect(url_for('login'))
+
+@app.route("/profile/", methods = ['GET', 'POST'])
+@login_required
+def profile():
+    if request.method == 'GET':
+        profile_details = _get_profile_information(session['user_id'])
+        # return str(profile_details[0][1])
+        return render_template('profile.html', first_name=profile_details[0][0], last_name=profile_details[0][1], birth_date=profile_details[0][2])
+    if request.method == 'POST':
+        username = request.form['username']
+        first_name = request.form['first_name']
+        last_name = request.form['last_name']
+        birth_date = request.form['birth_date']
+        _profile_update(first_name, last_name, birth_date)
+        return render_template('profile.html', first_name=first_name, last_name=last_name, birth_date=birth_date)
+
+# DROP TABLE if exists user;
```

Comment:
Whenever you create a pull request, the code is supposed to be ready to merge. Meaning that there shouldn't be unfinished, or partial things around. Having commented code for example is a bad sign that makes me think there's something still not finished.
Conclusion, make sure to remove all unnecessary comments before presenting the PR.